### ISSUE TITLE
Accuracy of 99% is achieved after ~40 epochs

### ISSUE THREAD

**Issue by hudongloop** [josephpmcdonald/tensorflowtrials#2]
hello,
when run the code of tf_lstm.py, But the test accuracy and train  accuracy  has 99%, what should adjust some parameter before run.
thanks.

**Comment by jpmcd** [comment]
You mean just the training accuracy and validation accuracy? The test accuracy is not checked until after all epochs have been completed, line 220 in tf_lstm.py. The training accuracy is high because the size of the training data is relatively small, but the validation accuracy (lines 217-218) should only be around 80%. There is no parameter to tune, wait for epoch number max_max_epoch to complete, or 75 epochs as written in my master branch, to see the test accuracy.

**Comment by hudongloop** [comment]
The test accuracy of finally is also 99% and the last two epochs is here:

`Epoch: 74 Learning rate: 0.032
batch size 20
Testing 1998 samples:
batch 98 / 99
Training Accuracy = 0.9995, time = 15.672 seconds

batch size 20
Testing 105 samples:
batch 4 / 5
Valid Accuracy = 0.8200

Epoch: 75 Learning rate: 0.031
batch size 20
Testing 1998 samples:
batch 98 / 99
Training Accuracy = 1.0000, time = 15.799 seconds

batch size 20
Testing 105 samples:
batch 4 / 5
Valid Accuracy = 0.8100

batch size 20
Testing 2103 samples:
batch 104 / 105
Test Accuracy = 0.9900`

Is our data different? My dateset is from [here](http://www-labs.iro.umontreal.ca/~lisa/deep/data/)

**Comment by jpmcd** [comment]
Our test datasets are different, note the different number of samples in my output printed in #3. My test data also came from that pickle file (note the method get_dataset() in imdb.py) but I'm not sure why yours is a different size. Since our validation accuracy is the same while test data matches the accuracy of the training data, are you sure you're not somehow mixing the training set with the test set? That would be the simplest explanation for the high accuracy. Perhaps the imdb file has changed since I downloaded that pickle file, which was over a year ago.

**Comment by jpmcd** [comment]
closing this
